### PR TITLE
27 add compact hover data to map pin

### PR DESCRIPTION
**Pull request by JoshuaMoelans**

### CODE REVIEW

**Comment by JoshuaMoelans** [comment]
Marked as draft until master is merged into the branch.

**Review by Scryper — APPROVED** [review]

File: components/VueLayerMarker.vue [hunk 1]
```diff
@@ -24,7 +37,54 @@ export default defineComponent({
...
+    },
+    emits: ["popup-toggled"],
+    data () {
+      return {
+        iconSrc: this.src,
+        selectediconSrc: this.selectSrc
+      };
+    },
+    methods: {
+      onSelect (e : any) {
```
Comment (jbelien) [inline components/VueLayerMarker.vue:58]:
```suggestion
      onSelect (e : SelectEvent) {
```

Good practice: avoid use of `any` as much as possible, that's the point of using TypeScript.
Change above is not enough, you will also need to import it I think (not tested).

File: components/VueLayerMarker.vue [hunk 2]
```diff
@@ -24,7 +37,54 @@ export default defineComponent({
...
+        // convert timestamp to readable format
+        const timestamp : Date = new Date(f.timestamp * 1000);
+        const tsString : string = timestamp.toLocaleString();
+        const details = { id: f.id, longitude: f.longitude, latitude: f.latitude, timestamp: tsString };
+        this.$root.$emit("popup-toggled", markerCoords, details);
+        /*
+         * console.log(this.displayDetails, this.details, this.coordinates);
+         * re-transform to avoid disappearing markers
+         */
+        e.feature.getGeometry().transform("EPSG:4326", "EPSG:3857");
```
Comment (jbelien) [inline components/VueLayerMarker.vue:83]: Using suggestion from previous comment should make this unnecessary.

File: components/VueLayerMarkerPopup.vue [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            this.details = details;
+      });
+      this.$root.$on("popup-hide", () => {
+        this.display = false;
+        this.position = [0, 0];
+      });
+    },
+    methods: {
+      toggleDetails () {
+        console.log("toggling details");
```
Comment (jbelien) [inline components/VueLayerMarkerPopup.vue:59]: Remove this `console.log`

File: plugins/vueLayers.js [hunk 4]
```diff
@@ -6,5 +6,7 @@
 import Vue from "vue";
 import VueLayers from "vuelayers";
 import "vuelayers/dist/vuelayers.css"; // needs css-loader
+import { SelectInteraction } from "vuelayers";
```
Comment (jbelien) [inline plugins/vueLayers.js:9]: Multiple import ; combine with line 7

File: components/VueLayerMarker.vue (hunk 1 above)
Comment (JoshuaMoelans) [reply components/VueLayerMarker.vue:58]: Resolved by adding a comment about the weird nature of the VueLayers event

File: components/VueLayerMarkerPopup.vue (hunk 3 above)
Comment (JoshuaMoelans) [reply components/VueLayerMarkerPopup.vue:59]: Removing it :)

File: components/VueLayerMarker.vue (hunk 2 above)
Comment (JoshuaMoelans) [reply components/VueLayerMarker.vue:83]: Indeed it does

File: plugins/vueLayers.js (hunk 4 above)
Comment (JoshuaMoelans) [reply plugins/vueLayers.js:9]: Done!

**Review by jbelien — APPROVED** [review]

**Pull request merged**